app/uk/gov/hmrc/hec/repos/HECTaxCheckStore.scala

```diff
@@ -66,76 +67,67 @@ trait HECTaxCheckStore {
 
 @Singleton
 class HECTaxCheckStoreImpl @Inject() (
-  mongo: ReactiveMongoComponent,
+  mongo: MongoComponent,
   configuration: Configuration
 )(implicit
   ec: ExecutionContext
-) extends HECTaxCheckStore
+) extends MongoCacheRepository(
+      mongoComponent = mongo,
+      replaceIndexes = true,
+      collectionName = "hecTaxChecks",
+      ttl = configuration.get[FiniteDuration]("hec-tax-check.ttl").value,
+      timestampSupport = new CurrentTimestampSupport(), // Provide a different one for testing
+      cacheIdType = CacheIdType.SimpleCacheId // Here, CacheId to be represented with `String`
+    )
+    with HECTaxCheckStore
     with Logging {
 
   val key: String                      = "hec-tax-check"
   private val ggCredIdField: String    = s"data.$key.taxCheckData.applicantDetails.ggCredId"
   private val isExtractedField: String = s"data.$key.isExtracted"
 
-  private val hecIndexes = Seq(
-    Index(Seq("ggCredId" -> IndexType.Ascending)),
-    Index(
-      Seq("isExtracted" -> IndexType.Ascending),
-      partialFilter = Some(BSONDocument("isExtracted" -> false))
+  //had to create default index on lastUpdated explicitly as it was not getting created .
+  //calling super.index was calling indexes field of PlayMongoRepository, which is final and hence was giving run time error.
+  def defaultIndexes: Seq[IndexModel] = Seq(
```

Comment:
would it work instead if you didn't have `defaultIndexes` here but just did this?
```scala
override def ensureIndexes: Future[Seq[String]] =
  super.ensureIndexes.flatMap(_ =>
     MongoUtils.ensureIndexes(collection, mongoNewIndexes, false))
```
That way we wouldn't need to copy the TTL index over here

Reply:
no, i haven't tried this. it seems it should work. let me try this

Reply:
it's working , updated